### PR TITLE
fix(audit_config): support audit config in 2.x domains/clusters

### LINKED ISSUE #69: [BUG]: audit config broken with 2.x beta
### What is the bug?

With `2.0.0-beta.1` of the provider, we can't manage our existing [Audit logs](https://opensearch.org/docs/2.9/security/access-control/api/#audit-logs) configuration anymore.

Terraform `plan/apply` fails with:

```
╷
│ Error: audit config only available from OpenSearch >= 1.0, got version 2.7.0
│ 
│   with module.logs.opensearch_audit_config.audit[0],
│   on ../../../modules/opensearch/main.tf line 257, in resource "opensearch_audit_config" "audit":
│  257: resource "opensearch_audit_config" "audit" {
│ 
╵
```

I guess this is related to this [version check](https://github.com/opensearch-project/terraform-provider-opensearch/blob/08ca809a1111164e3d8a2fa2d3bfd1ea11700a29/provider/resource_opensearch_audit_config.go#L202). AFAIK the `/_plugins/_security/api/audit` API is supported in `1.x` and `2.x` of OpenSearch.

### How can one reproduce the bug?

Create an `opensearch_audit_config` resource using the `1.x` version of this provider and then try to apply again using `2.0.0-beta.1` (guess creating such a resource with 2.x will fail as well).

### What is the expected behavior?

`opensearch_audit_config` resources can be managed with terraform against OpenSearch 2.x clusters/domains.

### What is your host/environment?

AWS OpenSearch domain using `2.7.0` of OpenSearch

```
{
    "name": "foo",
    "cluster_name": "bar",
    "cluster_uuid": "baz",
    "version": {
        "distribution": "opensearch",
        "number": "2.7.0",
        "build_type": "tar",
        "build_hash": "unknown",
        "build_date": "some-date",
        "build_snapshot": false,
        "lucene_version": "9.5.0",
        "minimum_wire_compatibility_version": "7.10.0",
        "minimum_index_compatibility_version": "7.0.0"
    },
    "tagline": "The OpenSearch Project: https://opensearch.org/"
}
```

### PR DESCRIPTION
**Pull request by moritzzimmer**
### Description

Adapted version check for audit config to support 2.x OpenSearch domains/clusters. 

### Issues Resolved

#69 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ func main() {
 
 	if debugMode {
 		//nolint:staticcheck // SA1019 ignore this!
-		err := plugin.Debug(context.Background(), "registry.terraform.io/opensearch-project/terraform-provider-opensearch",
+		err := plugin.Debug(context.Background(), "registry.terraform.io/opensearch-project/opensearch",
```
Comment (bbarani) [inline main.go:24]: I think the URL has to be https://registry.terraform.io/providers/opensearch-project/opensearch

Comment (moritzzimmer) [reply main.go:24]: This is actually unreleated to the fix, so I could leave out this change. 

Nevertheless terraform prints a warning, that providers should not be prefixed with `terraform-provider-` and using `registry.terraform.io/opensearch-project/opensearch` worked in our environment.

**Comment by moritzzimmer** [comment]
any updates?

File: main.go (hunk 1 above)
Comment (prudhvigodithi) [reply main.go:24]: Hey @moritzzimmer if its unrelated can you please change back to `registry.terraform.io/opensearch-project/terraform-provider-opensearch`? We can track this as a separate issue.

**Comment by prudhvigodithi** [comment]
Hey @moritzzimmer, thanks for waiting, please address this [comment](https://github.com/opensearch-project/terraform-provider-opensearch/pull/73#discussion_r1320281799), we can proceed to merge this PR.
Thanks

**Comment by moritzzimmer** [comment]
@prudhvigodithi I reverted the unnecessary change

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Thanks @moritzzimmer, LGTM.

**Pull request merged**